### PR TITLE
cmd/swarm: add password to ACT

### PR DESCRIPTION
**Pull request by acud**

### CODE REVIEW

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 95e184c551d5c9491e4e6428cf0a5cc2e4f21175
  * Commits must be in the following format: **%{scope}: %{description}**

 * Commit: 81feddbafefd0ebcf665b50f504e33fe5ad29411
  * Commits must be in the following format: **%{scope}: %{description}**

 * Commit: 6373b62c94f5624c8d9b7f74acdf78bb08e62c95
  * Commits must be in the following format: **%{scope}: %{description}**

Guidelines are available at https://github.com/ethersphere/go-ethereum

---
This message was auto-generated by https://gitcop.com

File: swarm/testutil/file.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+// it returns the complete file path
+func TempFileWithContent(t *testing.T, content string) string {
+	tempFile, err := ioutil.TempFile("", "swarm-temp-file")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	_, err = io.Copy(tempFile, strings.NewReader(content))
+	if err != nil {
+		t.Fatal(err)
```
Comment (lmars) [inline swarm/testutil/file.go]: Remove the tmp file if there is an error.

File: swarm/api/act.go [hunk 2]
```diff
@@ -383,7 +401,7 @@ func DoPKNew(ctx *cli.Context, privateKey *ecdsa.PrivateKey, granteePublicKey st
 	return sessionKey, ae, nil
 }
 
-func DoACTNew(ctx *cli.Context, privateKey *ecdsa.PrivateKey, salt []byte, grantees []string) (accessKey []byte, ae *AccessEntry, actManifest *Manifest, err error) {
+func DoACTNew(ctx *cli.Context, privateKey *ecdsa.PrivateKey, salt []byte, grantees []string, encryptPassword string) (accessKey []byte, ae *AccessEntry, actManifest *Manifest, err error) {
 	if len(grantees) == 0 {
```
Comment (zelig) [inline swarm/api/act.go]: now that we allow passwords, we should not return if no grantees via pubkey. only if both are empty no?

File: cmd/swarm/access.go [hunk 3]
```diff
@@ -110,23 +110,33 @@ func accessNewACT(ctx *cli.Context) {
...
-	accessKey, ae, actManifest, err = api.DoACTNew(ctx, privateKey, salt, grantees)
+
+	if encryptPasswordFilename != "" {
+		bytes, err = ioutil.ReadFile(encryptPasswordFilename)
+		if err != nil {
+			utils.Fatalf("could not read password filename: %v", err)
+		}
+		encryptPassword = string(bytes)
+	}
+	accessKey, ae, actManifest, err = api.DoACTNew(ctx, privateKey, salt, grantees, encryptPassword)
```
Comment (zelig) [inline cmd/swarm/access.go]: I dont get this. Is the file not supposed to contain passwords one per line analogous to grantees file?
I cannot find the logic of split and iterate like it is on 138 for grantees....

File: swarm/api/act.go (hunk 2 above)
Comment (acud) [reply swarm/api/act.go]: yes

File: cmd/swarm/access.go (hunk 3 above)
Comment (acud) [reply cmd/swarm/access.go]: it wasn't crystal clear that it was password**s** and not passwor**d** :)
in any case, in regard also to your previous comment - i'm implicitly assuming we should accommodate a scenario where just passwords are provided and no grantee public keys are provided

File: cmd/swarm/access.go [hunk 4]
```diff
@@ -110,23 +110,33 @@ func accessNewACT(ctx *cli.Context) {
...
 
 	bytes, err := ioutil.ReadFile(actFilename)
 	if err != nil {
 		utils.Fatalf("had an error reading the grantee public key list")
 	}
 	grantees = strings.Split(string(bytes), "\n")
-	accessKey, ae, actManifest, err = api.DoACTNew(ctx, privateKey, salt, grantees)
+
+	if encryptPasswordFilename != "" {
+		bytes, err = ioutil.ReadFile(encryptPasswordFilename)
```
[the remark this answers is not in the document]
Comment (acud) [reply cmd/swarm/access.go]: @lmars obsoleted by @zelig's comments - this has to accommodate a password list input

File: swarm/api/act.go [hunk 5]
```diff
@@ -261,18 +267,29 @@ func (a *API) doDecrypt(ctx context.Context, credentials string, pk *ecdsa.Priva
...
 		case "act":
-			publisherBytes, err := hex.DecodeString(m.Access.Publisher)
-			if err != nil {
-				return ErrDecrypt
-			}
-			publisher, err := crypto.DecompressPubkey(publisherBytes)
-			if err != nil {
-				return ErrDecrypt
-			}
+			var (
```
Comment (zelig) [inline swarm/api/act.go]: not sure what behaviour this code has. 
What we need is to try generate the session key using pk for the current bzzaccount. 
If this fails, we fall back to password

**Comment by acud** [comment]
@zelig done

**Comment by acud** [comment]
reopening on upstream

**Pull request closed without merging**